Change the `stackAllItems` option to be `false` by default (#1645)
sindresorhus/got: readme.md

The function takes three arguments:
- `response` - The current response object.

> - `allItems` - An array of the emitted items if [pagination.stackAllItems](#pagination.stackAllItems) is set to `true`. An empty array otherwise.

Reviewer: Can you move this property to the bottom of the list. Same in the types.

Can you make it clear that it's empty by default?

Author: Sure. Just for me to remember how to correctly update a branch from upstream... My git proficiency is rather basic.

Reviewer 2: `git pull --rebase origin branch`

Author: Better?

The function takes an object with the following properties:
- `response` - The current response object.
- `currentItems` - Items from the current response.

> - `allItems` - An empty array, unless when [pagination.stackAllItems](#pagination.stackAllItems) is set to `true`. In the later case, an array of the emitted items.

Reviewer 2: ```suggestion
- `allItems` - An empty array, unless when [`pagination.stackAllItems`](#pagination.stackAllItems) is set to `true`. In the later case, an array of the emitted items.
```

> - `allItems` - An empty array, unless when [`pagination.stackAllItems`](#pagination.stackAllItems) is set to `true`. In the later case, an array of the emitted items.

Reviewer: ```suggestion
- `allItems` - An empty array, unless [`pagination.stackAllItems`](#pagination.stackAllItems) is set to `true`, in which case, it's an array of the emitted items.
```

###### pagination.stackAllItems

Type: `boolean`\
Default: `false`

> Defines how the property `allItems` in [pagination.paginate](#pagination.paginate), [pagination.filter](#pagination.filter) and [pagination.shouldContinue](#pagination.shouldContinue) is managed.

Reviewer 2: ```suggestion
Defines how the property `allItems` in [`pagination.paginate`](#pagination.paginate), [`pagination.filter`](#pagination.filter) and [`pagination.shouldContinue`](#pagination.shouldContinue) is managed.
```

> 
> By default, the property `allItems` is always an empty array.

Reviewer 2: ```suggestion
```

> By default, the property `allItems` is always an empty array. This setting can be helpful to save on memory usage when working with a large dataset.

Reviewer: The second sentence makes less sense now that it's `false` by default. It should be formulated differently.

Author: Done

> When set to `false`, the parameter `allItems` is always an empty array. This setting can impact memory usage when working with a large dataset.

Reviewer 2: ```suggestion
When set to `false`, the `allItems` parameter is always an empty array. If `true`, it can hugely increase memory usage when working with a large dataset.
```

> When set to `true`, the property `allItems` is an array of the emitted items.

Reviewer 2: ```suggestion
When set to `false`, the parameter `allItems` is always an empty array.
```
